mlir-tensorrt/compiler/lib/Dialect/TensorRTRuntime/IR/TensorRTRuntime.cpp

```diff
@@ -113,6 +113,62 @@ void EnqueueOp::getEffects(
   }
 }
 
+//===----------------------------------------------------------------------===//
+// EnqueueAllocOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult EnqueueAllocOp::verify() {
+  // Verify host tensor indices.
+  if (std::optional<ArrayRef<int64_t>> hostTensorIndices =
+          getHostTensorArgs()) {
+    // We don't count the context and stream argument here.
+    const int64_t numInputArgs = getInputs().size();
+    for (int64_t idx : *hostTensorIndices) {
+      if (idx >= numInputArgs || idx < 0)
+        return emitOpError("host_tensor_args value ")
+               << idx << " is out of bounds";
+      Value operand = getInputs()[idx];
+      Type elType = mlir::getElementTypeOrSelf(operand.getType());
+      if (!elType.isInteger(32))
+        return emitOpError("host tensor arguments must have element type i32, "
+                           "but input arg ")
+               << idx << " has type " << operand.getType();
+    }
+  }
+
+  // Verify that all results are either all tensors or all memrefs
+  if (getNumResults() > 0) {
+    bool allTensors = getResult(0).getType().isa<TensorType>();
+    for (auto result : getResults()) {
+      if (result.getType().isa<TensorType>() != allTensors) {
+        return emitOpError("all results must be of the same type (all tensors "
+                           "or all memrefs)");
+      }
+    }
+  }
+  return success();
+}
+
+void EnqueueAllocOp::getEffects(
+    SmallVectorImpl<SideEffects::EffectInstance<MemoryEffects::Effect>>
+        &effects) {
+  // This op allocates memory for its results
```

Comment:
Memory effects are restricted to the case where the op has all memref types. If operands/results have tensor types, then just return from this function before modifying `effects` (don't add allocation effect either).